### PR TITLE
Add io.github.pejuko.follamac

### PR DESCRIPTION
**Pull request by pejuko**
### Please confirm your submission meets all the criteria

Please describe your application briefly below.

Follamac is a desktop application which provides convenient way to work with Ollama and large language models (LLMs).

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing

Comment (hfiguiere) [reply flathub.json:2]: since it's built from source why limiting?

Comment (pejuko) [reply flathub.json:2]: I have no opportunity to test it on different architectures :-(

Comment (pejuko) [reply flathub.json]: ok, I removed the limitation to x86_64

**Comment by bbhtt** [comment]
bot, build io.github.pejuko.follamac

File: io.github.pejuko.follamac.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        "env": {
+          "XDG_CACHE_HOME": "/run/build/follamac/flatpak-node/cache",
+          "YARN_CACHE_FOLDER": "/run/build/follamac/flatpak-node/npm-cache"
+        }
+      },
+      "build-commands": [
+        "HOME=$PWD yarn config --offline set yarn-offline-mirror $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror",
+        "yarn install --offline",
+        "yarn --offline build",
+        "yarn --offline dist --x64 --linux --dir",
```
Comment (hfiguiere) [inline io.github.pejuko.follamac.json:39]: if you pass `x64` then for sure things won't work on anything else.

Comment (pejuko) [reply io.github.pejuko.follamac.json]: should be fixed

**Comment by bbhtt** [comment]
bot, build io.github.pejuko.follamac

**Comment by bbhtt** [comment]
Seeing some kind of connection problem

image

**Comment by pejuko** [comment]
You need Ollama ( https://ollama.com/ ) running on your localhost then restart follamac or click the reload button and pull some model. The model can be then pulled from follamac or from command line. From command line type something like: 

ollama pull llama3

If you wish to pull from follamac you can write llama3 to "Model name to pull" input box and click the PULL button.

**Comment by bbhtt** [comment]
Ok thanks. Can you please add these instructions as a note in the first line of description tag in appadata file? 


image


Or preferably as a dialogue box on first start. This can be done later.

**Comment by bbhtt** [comment]
The text generation seems a bit laggy. It might be due to being in a VM. otherwise seems ok to me after the last comment is fixed.


https://github.com/flathub/flathub/assets/62639087/9f717e69-3670-4d7a-9de4-711b2e69d46a

**Comment by bbhtt** [comment]
bot, build io.github.pejuko.follamac

**Comment by pejuko** [comment]
Ok, I put the instructions into the description.
The lagginess really depends on the computer, GPU, model size, ...

**Pull request closed without merging**